### ISSUE TITLE
ID fields come back null

### ISSUE THREAD

**Issue by berinle** [VanRoy/spring-data-jest#24]
ID field comes back null despite annotating id field with `@JestId` per documentation at https://github.com/searchbox-io/Jest/tree/master/jest#indexing-documents?

The `_id` generated by ES is never set on the entity id.

Please advice.

**Comment by VanRoy** [comment]
Hi @berinle, could send me more details about your issue ?

Which version you use ?
Can you send me a code snippets to reproduce ?

Thanks.
Julien.

**Comment by berinle** [comment]
Hi @VanRoy, 
I have am able to reproduce this on versions 2.2.0 and 2.3.0. I just forked your sample, made changes and tested with that.

Here are the changes I made: https://github.com/berinle/spring-data-jest/commit/8480ab23cfd537fe43c4cfe133667f3b16693cf7

You can clone and test. You will notice despite the `@JestId` annotation on the `id` property, the results are still coming back null.

```
....
2017-02-22 11:27:35.242  INFO 1443 --- [           main] o.s.j.e.a.AnnotationMBeanExporter        : Registering beans for JMX exposure on startup
2017-02-22 11:27:35.374  INFO 1443 --- [pdateTask][T#1]] org.elasticsearch.cluster.metadata       : [Pete Wisdom] [customer] update_mapping [customer]
Customers found with findAll():
-------------------------------
Customer[id=null, firstName='Alice', lastName='Smith']
Customer[id=null, firstName='Bob', lastName='Smith']

Customer found with findByFirstName('Alice'):
--------------------------------
Customer[id=null, firstName='Alice', lastName='Smith']
Customers found with findByLastName('Smith'):
--------------------------------
Customer[id=null, firstName='Alice', lastName='Smith']
Customer[id=null, firstName='Bob', lastName='Smith']
```

Please let me know if you need any other info from me. Thanks

**Comment by VanRoy** [comment]
@berinle Indeed only @org.springframework.data.annotation.Id is supported to keep close of Spring Data Elasticsearch.

Have you any problem to use @org.springframework.data.annotation.Id instead of @JestId ?

**Comment by berinle** [comment]
@vanroy, @id isn't working as well. That's why I tried @jestid instead. You can confirm that yourself, just running your sample. 

The customer entries printed on the screen all come back with null ids

**Comment by VanRoy** [comment]
Hi @berinle , indeed the two options doesn't work. It's due to bug on  my autoconfiguration.
I will fix it quickly on next release.

You can use this workaround awaiting the fix : 

Add this snippets on your project ( on @Configuration annoted class )

```
@Bean
public ElasticsearchOperations elasticsearchTemplate(JestClient client) {
	return new JestElasticsearchTemplate(client);
}
```

**Comment by berinle** [comment]
@VanRoy cool! I'll apply the workaround for now. Thanks!

**Comment by gkrs** [comment]
Hi @VanRoy
I came across the same issue. I have tried the workaround, even applied your fix but it doesn't help.
What works for me is to have the elasticTemplate bean created this way:
```java
@Bean
@ConditionalOnMissingBean
public ElasticsearchOperations elasticsearchTemplate(JestClient client, ElasticsearchConverter elasticsearchConverter, SimpleElasticsearchMappingContext mappingContext) {
	return new JestElasticsearchTemplate(client, elasticsearchConverter, new DefaultJestResultsMapper(mappingContext));
}
```

**Comment by VanRoy** [comment]
Hi @gkrs , can you try the last snapshot version available on master branch and tell me if it fix your issue ?

Thanks for you help.

**Comment by gkrs** [comment]
@VanRoy I've tried the latest master. The issue still exists. I receive "PersistentEntity must not be null!" message.

**Comment by VanRoy** [comment]
Hi @gkrs , your issue seems not to be the same. Can you send me more informations ( stacktrace, sample project ), i will investigate on it.

**Comment by VanRoy** [comment]
Closed with release 2.3.1.

@gkrs , please if you can reproduce your bug, open a new issue with more informations.
Thanks for you help.
